### PR TITLE
Add agentic SFT training objective contract

### LINKED ISSUE #689: OpenSearch-VL alignment: support agentic LoRA SFT - Milestone 2: train LoRA adapters from agentic traces - Unit 1: Extend `LoRATrainingConfig` and dataset validation for `training_objective=agentic_sft`.
## Executable Unit

Extend `LoRATrainingConfig` and dataset validation for `training_objective=agentic_sft`.

## Parent Context

- Direction: `OpenSearch-VL alignment: support agentic LoRA SFT`
- Plan/Milestone: `Milestone 2: train LoRA adapters from agentic traces`

## Definition Of Done

- The change is scoped to the smallest useful implementation slice.
- The governing spec or plan is created or updated before broad code changes.
- Tests cover the new behavior or contract edge case.
- Metrics and verification output are recorded in the PR evidence.

## Suggested Verification

- `git diff --check`
- Focused pytest or Swift test for the touched code path
- Changed-scope coverage for modified Python or Swift files when measurable
- Any relevant benchmark/evaluation artifact if this changes measured behavior

### PR DESCRIPTION
**Pull request by Zigfreidish**
## Summary
- Add the explicit `training_objective=agentic_sft` contract for LoRA SFT over `agentic_tool_trace` datasets.
- Validate incompatible objective overrides and agentic trace sample metadata before backend training starts.
- Update the agentic trajectory contract, active plan, and LoRA workflow runbook with the objective and dataset-contract behavior.

Fixes #689.

## Plan or Spec
- `docs/plans/2026-05-20-agentic-lora-sft-formatting.md`
- `docs/agentic-trajectory-dataset-contract.md`
- `docs/runbooks/phase-8-lora-adapter-workflow.md`

## Commands Run
```text
git diff --check
# pass

git diff --cached --check
# pass

PYTHONPATH="$(pwd):$(pwd)/services/mlx-worker-python" uv run --project services/mlx-worker-python pytest services/mlx-worker-python/tests/test_lora_model_ops.py services/mlx-worker-python/tests/test_lora_model_ops_unit.py services/mlx-worker-python/tests/test_trajectory_provenance.py services/mlx-worker-python/tests/test_training_dataset_builder.py services/mlx-worker-python/tests/test_agentic_sft_formatter.py -q
# 293 passed, 2 warnings in 4.19s

PYTHONPATH="$(pwd):$(pwd)/services/mlx-worker-python" uv run --project services/mlx-worker-python coverage run --include='*/worker/model_ops/training_config.py,*/worker/model_ops/lora_training_pipeline.py,*/worker/model_ops/training_dataset.py,*/worker/model_ops/agentic_sft_formatter.py' -m pytest services/mlx-worker-python/tests/test_lora_model_ops.py services/mlx-worker-python/tests/test_lora_model_ops_unit.py services/mlx-worker-python/tests/test_trajectory_provenance.py services/mlx-worker-python/tests/test_training_dataset_builder.py services/mlx-worker-python/tests/test_agentic_sft_formatter.py -q
# 293 passed, 2 warnings

PYTHONPATH="$(pwd):$(pwd)/services/mlx-worker-python" uv run --project services/mlx-worker-python coverage report --include='*/worker/model_ops/training_config.py,*/worker/model_ops/lora_training_pipeline.py,*/worker/model_ops/training_dataset.py,*/worker/model_ops/agentic_sft_formatter.py' --show-missing
# TOTAL 99%

python3 - <<'PY'
from pathlib import Path
from scripts.pre_commit_gate import pre_commit_base_ref, staged_changed_files, run_performance_report
root = Path.cwd()
base = pre_commit_base_ref(root)
changed = staged_changed_files(root, base_ref=base)
print(f"base_ref={base}")
print(f"changed_files={len(changed)}")
for path in changed:
    print(f"changed={path}")
outcome = run_performance_report(root, changed, base_ref=base)
print(f"status={outcome.status}")
print(f"report_dir={outcome.report_dir}")
raise SystemExit(0 if outcome.status == "ok" else 1)
PY
# Status: ok; 2 direct/gated probes; regressions 0; verification failures 0
# Report: .runtime/pre-commit-performance/20260519-233221-53582b75/report/report.md

make py-test
# 2854 passed, 14 skipped, 2 warnings in 127.48s

git commit -m "Add agentic SFT training objective contract"
# pre-commit hook completed:
# make swift-test passed
# make py-test: 2854 passed, 14 skipped, 2 warnings in 135.36s
# make integration-test: 114 passed, 1 skipped in 643.40s
# PR-scoped performance: Status ok; 2 direct/gated probes; regressions 0; verification failures 0
# Hook report: .runtime/pre-commit-performance/20260519-235825-53582b75/report/report.md
```

## Coverage and Metrics
- Changed-scope coverage report total: 99%.
- File coverage: `agentic_sft_formatter.py` 100%, `lora_training_pipeline.py` 100%, `training_config.py` 97%, `training_dataset.py` 99%.
- Pre-commit changed-line coverage:
  - `training_config.py`: 100.00%
  - `lora_training_pipeline.py`: 100.00%
  - `test_lora_model_ops.py`: 100.00%
- PR-scoped performance report: `Status: ok`, direct/gated probes 2, regressions 0, verification failures 0.
- Pre-commit performance metrics:
  - Training config target-module cache: elapsed -1.30%, peak bytes neutral, coverage pass 100.0%.
  - LoRA reward summary candidate min/max reuse: elapsed +0.50%, sorted calls unchanged, coverage pass 100.0%.
- Observability mode: N/A, this change adds config and dataset validation only; no runtime observability or probe collection path changed.
- Probe overhead: N/A for production serving; only existing PR-scoped offline probes ran.

## Known Gaps
None.

## Evidence Checklist
- [x] Relevant plan or spec is identified.
- [x] Behavior changes are reflected in the relevant docs.
- [x] Protocol changes include regenerated generated artifacts.
- [x] Dependency changes include updated lockfiles.
- [x] Relevant tests were run.
- [x] A metrics report is included, or `N/A` is stated explicitly with the reason.
- [x] Observability mode, probe overhead, and any deferred debug or evidence work are recorded, or `N/A` is stated explicitly with the reason.
- [x] Deferred work and known gaps are stated explicitly.

### CODE REVIEW

File: services/mlx-worker-python/tests/test_lora_model_ops.py [hunk 1]
```diff
@@ -49,6 +49,82 @@
 from worker.runtime.mlx_text_runtime import MLXTextRuntime, RuntimeTokenEvent
 
 
+@pytest.fixture(autouse=True)
+def _verify_agentic_sft_training_contract() -> None:
```
[the remark this answers is not in the document]
Comment (Zigfreidish) [reply services/mlx-worker-python/tests/test_lora_model_ops.py]: Fixed in bd07cdd3. I moved the agentic SFT contract assertions out of the autouse fixture and into services/mlx-worker-python/tests/test_agentic_sft_training_contract.py, so the LoRA model-ops suite no longer runs the contract checks before every test. Re-ran focused tests, changed-scope coverage, full pre-commit gate, and PR-scoped performance after the change.
